CRM/RcBase/Api/Get.php

```diff
@@ -0,0 +1,335 @@
+            ->execute();
+
+        return self::parseResultsFirst($results, 'id');
+    }
+
+    /**
+     * Get Relationship ID from contact and phone type
+     *
+     * @param int $contact_id Contact ID
+     * @param int $other_contact_id Other contact ID (of the relation)
+     * @param int $relationship_type_id Relationship type ID
+     * @param bool $check_permissions Should we check permissions (ACLs)?
+     *
+     * @return int|null Relationship ID if found, null if not found
+     *
+     * @throws API_Exception
+     * @throws UnauthorizedException
+     * @throws CRM_Core_Exception
+     */
+    public static function relationshipID(
+        int $contact_id,
+        int $other_contact_id,
+        int $relationship_type_id,
+        bool $check_permissions = false
+    ): ?int {
+
+        if ($contact_id < 1 || $other_contact_id < 1 || $relationship_type_id < 1) {
+            throw new CRM_Core_Exception('Invalid ID.');
+        }
+
+        $results = Relationship::get($check_permissions)
+            ->addSelect('id')
+            ->addWhere('contact_id_a', '=', $contact_id)
+            ->addWhere('contact_id_b', '=', $other_contact_id)
+            ->addWhere('relationship_type_id', '=', $relationship_type_id)
+            ->setLimit(1)
+            ->execute();
+
+        return self::parseResultsFirst($results, 'id');
+    }
+
+    /**
+     * Get ID of default Location type
+     *
+     * @param bool $check_permissions Should we check permissions (ACLs)?
+     *
+     * @return int|null Location type ID if found, null if not found
+     *
+     * @throws API_Exception
+     * @throws UnauthorizedException
+     */
+    public static function defaultLocationTypeID(bool $check_permissions = false): ?int
+    {
+        $results = LocationType::get($check_permissions)
+            ->addSelect('id')
+            ->addWhere('is_default', '=', true)
+            ->setLimit(1)
+            ->execute();
+
+        return self::parseResultsFirst($results, 'id');
+    }
+
+    /**
+     * Get All Activity for a contact
+     *
+     * @param int $contact_id Contact ID
+     * @param int $record_type_id Contact role in activity (source, target, assignee)
+     * @param int $activity_type_id Optionally filter activities by this type
+     * @param bool $check_permissions Should we check permissions (ACLs)?
+     *
+     * @return array Array of Activity IDs
+     *
+     * @throws API_Exception
+     * @throws UnauthorizedException
+     * @throws CRM_Core_Exception
+     */
+    public static function allActivity(
+        int $contact_id,
+        int $record_type_id,
+        int $activity_type_id = 0,
+        bool $check_permissions = false
+    ): array {
+        $activities = [];
+
+        if ($contact_id < 1 || $record_type_id < 1 || $activity_type_id < 0) {
+            throw new CRM_Core_Exception('Invalid ID.');
+        }
+
+        // record_type_id=3 means contact is the target of activity
```

Comment:
This comment might be unnecessary here. It could be more useful at that line where this function is called with record_type_id = 3 value.

Reply:
removed